### DISCUSSION TITLE
Cannot retrieve my private key from environment variables

### DISCUSSION THREAD

**Discussion by yarg119** [smartcontractkit/full-blockchain-solidity-course-py#1194] [Q&A] [answered]
It seems like I have done everything right. I have put in my environment variable name as "PRIVATE_KEY" and the value as "PRIVATE_KEY_Value(i.e) 0x......" When I run deploy.py, it will not print my private key in the terminal. I have imported os. I use `private_key = os.environ.get("PRIVATE_KEY")` to retrieve the key followed by `print(private_key)` In return, I get `INFO: Could not find files for the given pattern(s).
None` What the hell is the problem with this. It is driving me crazy, and I have gotten the same answers from Stack Exchange Ethereum and Youtube.

**Comment by yarg119** [comment, marked as answer]
Got it figured out. I had to make  a .env file with `PRIVATE_KEY=0x......................` in my project to retrieve it.

**Reply by cromewar** [reply]
Nice Catch!
